pkg/pool-weighted/contracts/smart/InvestmentPool.sol

```diff
@@ -88,19 +89,44 @@ contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
         uint256 numTokens = tokens.length;
         InputHelpers.ensureInputLengthMatch(numTokens, normalizedWeights.length, assetManagers.length);
 
-        _tokenCountCache = numTokens;
+        _setMiscData(_getMiscData().insertUint7(numTokens, _TOTAL_TOKENS_OFFSET));
```

Comment:
We don't _need_ it, but it won't hurt to have a `require(numTokens < 128)` (due to the 7 bits), just in case.

Reply:
Maybe: _require(_getMaxTokens() == numTokens, Errors.MAX_TOKENS);
101-127 would still be an overflow. If you get the same value out that you put in, you know it didn't overflow

Reply:
A bit indirect, but I like it. Go for it!